### PR TITLE
feature/exam indicator

### PR DESCRIPTION
**Pull request by MarekMacko**
### Exam indicator depending on status reservation
image

### CODE REVIEW

File: src/main/java/pl/edu/zut/mad/schedule/ui/VerticalTextView.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import android.content.Context
+import android.graphics.Canvas
+import android.util.AttributeSet
+import android.view.Gravity
+import android.widget.TextView
+
+class VerticalTextView(context: Context, attrs: AttributeSet) : TextView(context, attrs) {
+
+    private val topDown: Boolean =
+        if (Gravity.isVertical(gravity) && (gravity and Gravity.VERTICAL_GRAVITY_MASK) == Gravity.BOTTOM) {
```
Comment (sswierczek) [inline src/main/java/pl/edu/zut/mad/schedule/ui/VerticalTextView.kt]: Maybe assign to local value before if and then return this value

File: src/main/java/pl/edu/zut/mad/schedule/ui/VerticalTextView.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    override fun onDraw(canvas: Canvas) {
+        val textPaint = paint
+        textPaint.color = currentTextColor
+        textPaint.drawableState = drawableState
+
+        canvas.save()
+
+        if (topDown) {
+            canvas.translate(width.toFloat(), 0F)
+            canvas.rotate(90F)
```
Comment (sswierczek) [inline src/main/java/pl/edu/zut/mad/schedule/ui/VerticalTextView.kt]: I would name this as constants rotate_clockwise_90 etc

File: src/test/java/pl/edu/zut/mad/schedule/MockData.kt [hunk 3]
```diff
@@ -106,7 +116,7 @@ internal class MockData {
         private fun getLessonUiWithCancellation(): LessonUi {
             val date = getUiDate()
             return LessonUi(SUBJECT, TYPE, ROOM, TEACHER_UI, FACULTY_ABBREVIATION, FIELD_OF_STUDY,
```
Comment (sswierczek) [inline src/test/java/pl/edu/zut/mad/schedule/MockData.kt:118]: Looks like it needs builder

**Review by sswierczek — APPROVED** [review]

File: src/test/java/pl/edu/zut/mad/schedule/MockData.kt (hunk 3 above)
Comment (MarekMacko) [reply src/test/java/pl/edu/zut/mad/schedule/MockData.kt:118]: It looks that I have to add some boilerplate to get builder. I searching for some libraries for generating this specially for Kotlin but I didn't find anyting.

**Pull request merged**